### ISSUE TITLE
GitCrate.get_package_directory() doesn't handle version.workspace

### ISSUE THREAD

**Issue by AdelKS** [projg2/pycargoebuild#22]
Hello,

I was trying to use `pycargoebuild` on this project: https://github.com/Rikorose/DeepFilterNet/tree/main/ladspa and it was aborting on this error
```
ValueError: No version found in /var/cache/distfiles/hdf5-rust-26046fb4900ec38afd2a1c0494cff688b288662e.gh.tar.gz
```
The  `Cargo.toml` is the project [is this one](https://github.com/aldanor/hdf5-rust/blob/26046fb4900ec38afd2a1c0494cff688b288662e/Cargo.toml), is something wrong with it or is it a bug ?

Thanks!

**Comment by AdelKS** [comment]
Okay I took time to debug a little what's going on and actually it's simply that the project https://github.com/aldanor/hdf5-rust has a subfolder where there's a [Cargo.toml which doesn't have a version](https://github.com/aldanor/hdf5-rust/blob/26046fb4900ec38afd2a1c0494cff688b288662e/hdf5-derive/Cargo.toml), but it looks like some kind of helper subproject, is it something that upstream needs to fix ?

**Comment by mgorny** [comment]
@CyberTailor, want to take a look at this, perhaps? This is somewhat related to what you worked on previously.

**Comment by MrRoy** [comment]
I can still reproduce this with pycargoebuild 0.13.0, with eg. https://github.com/signalapp/libsignal
```
# pycargoebuild rust/attest
[...]
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.11/pycargoebuild", line 8, in <module>
    sys.exit(entry_point())
             ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/__main__.py", line 384, in entry_point
    sys.exit(main(*sys.argv))
             ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/__main__.py", line 331, in main
    ebuild = get_ebuild(
             ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/ebuild.py", line 211, in get_ebuild
    crate_licenses=get_crate_LICENSE(crates, distdir, license_overrides),
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/ebuild.py", line 152, in get_crate_LICENSE
    crate_licenses = set(get_license_from_crate(crate, distdir)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/ebuild.py", line 152, in <genexpr>
    crate_licenses = set(get_license_from_crate(crate, distdir)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/ebuild.py", line 116, in get_license_from_crate
    base_dir = crate.get_package_directory(distdir)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/cargo.py", line 103, in get_package_directory
    metadata = get_package_metadata(
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/cargo.py", line 235, in get_package_metadata
    raise ValueError(f"No version found in {f.name}")
ValueError: No version found in /tmp/boring-8245063ae6eb97d909982b89fad45bb7f0a2a1a0.gh.tar.gz
```
[Here](https://github.com/signalapp/boring/blob/8245063ae6eb97d909982b89fad45bb7f0a2a1a0/Cargo.toml) is the Cargo.toml for that commit

**Comment by mgorny** [comment]
Sigh. That's because the archive doesn't contain `Cargo.lock`. I guess we need a better logic for finding the workspace root.

**Comment by mgorny** [comment]
Fixed now, in 0.13.1. I'm going to push it to ::gentoo shortly.

**Comment by MrRoy** [comment]
Thanks, I can confirm that it works with the above project on 0.13.1
